runbooks/mcp-to-permissioned-fps.md

```diff
@@ -0,0 +1,382 @@
+
+`op-proposer` has already been updated to support `op-contracts/v1.6.0` if you are using the latest governance-approved version of the OP Stack. You can safely run an instance of `op-proposer` for `op-contracts/v1.6.0` in parallel with your current production instance.
+
+**Configuration**
+
+You can generally use the same configuration that you currently use for `op-proposer` with the following modifications:
+
+- `L2OOAddressFlag`
+    - `--l2oo-address, L2OO_ADDRESS`
+    - Must be empty
+- `DisputeGameFactoryAddressFlag`
+    - `--game-factory-address, GAME_FACTORY_ADDRESS`
+    - Must be set to the address of the `DisputeGameFactoryProxy` contract
+- `DisputeGameTypeFlag`
+    - `--game-type, GAME_TYPE`
+    - Must be set to `1`
+- `ProposalIntervalFlag`
+    - `--proposal-interval, PROPOSAL_INTERVAL`
+    - Recommend setting this to `1h`
+
+**Running a challenger**
+
+Refer to the [Releases page](https://github.com/ethereum-optimism/optimism/releases) on the Optimism Monorepo to determine the latest OP Stack release that you should be using.
+
+> [!NOTE]
+> `op-challenger` v1.1.2 modified the `op-challenger` to disable prestate checks for the permissioned game. This change ensures that the `op-challenger` will correctly resolve games regardless of the configured prestate value.
+
+`op-challenger` is a service that participates in the dispute game process and challenges invalid proposals. The `op-challenger` will NOT have permission to post counterclaims in the permissioned games as the EOA account it uses is not the CHALLENGER role. Under this permissioned setup, the `op-challenger` primarily serves to resolve dispute games on behalf of the proposer and can act as additional monitoring. Refer to [the Optimism Developer Docs](https://docs.optimism.io/builders/chain-operators/tools/op-challenger) for a detailed overview of how to run the `op-challenger`.
+
+**Configuration**
+
+For networks not in the [superchain-registry](https://github.com/ethereum-optimism/superchain-registry/blob/main/chainList.json) you need: 
+
+- `CannonRollupConfigFlag`
+    - `cannon-rollup-config, CANNON_ROLLUP_CONFIG`
+    - Rollup chain parameters (cannon trace type only)
+- `CannonL2GenesisFlag`
+    - `cannon-l2-genesis, CANNON_L2_GENESIS`
+    - Path to the `op-geth` genesis file (cannon trace type only)
+- `CannonPreStateFlag`
+    - `cannon-prestate, CANNON_PRESTATE`
+    - Path to absolute prestate to use when generating trace data (cannon trace type only)
+    - **Important details**
+        - Must be `0x038512e02c4c3f7bdaec27d00edf55b7155e0905301e1a88083e4e0a6764d54c`. This is the same value as OP Mainnet because it should correspond to the latest version of the `op-program`. This version of the op-program doesn’t account for chains recently added or not in the `superchan-registry`, but for permissioned games that is fine because it will never actually execute.
```

Comment:
Just noticed this is likely confusing.  The absolute prestate hash from the contracts must be set to `0x038512e02c4c3f7bdaec27d00edf55b7155e0905301e1a88083e4e0a6764d54c` but the challenger `--cannon-prestate` option is actually a file path to the preimage of that state commitment. For the permissioned game it isn't actually loaded and can be set to anything even if the file doesn't exist, but it's potentially confusing to suggest it be set to the commit hash.  You can build the actual state for that state commitment with:

```
git clone https://github.com/ethereum-optimism/optimism -b op-program/v1.3.1 --recurse-submodules
cd optimism
make reproducible-prestate
```
The built prestate will be in `./op-program/bin/prestate.json`.  To be fully correct, the `cannon-prestate` option should point to that file (but it won't actually get used in permissioned games).